include/seqan3/io/sequence_file/format_embl.hpp

```diff
@@ -245,34 +255,51 @@ class format_embl
         }
         else
         {
-            if (ranges::empty(sequence)) //[[unlikely]]
+            if (std::ranges::empty(sequence)) //[[unlikely]]
                 throw std::runtime_error{"The SEQ field may not be empty when writing embl files."};
 
-            std::ranges::copy(std::string_view{"SQ Sequence "}, stream_it);
-            auto res = std::to_chars(&buffer[0], &buffer[0] + sizeof(buffer), sequence_size);
-            std::copy(&buffer[0], res.ptr, stream_it);
-            std::ranges::copy(std::string_view{" BP;\n"}, stream_it);
-            auto seqChunk = sequence | ranges::view::chunk(60);
-            unsigned int i = 0;
-            size_t bp = 0;
-            for (auto chunk : seqChunk)
+            // write beginning of sequence record
+            stream_it.write_range(std::string_view{"SQ Sequence "});
+            write_number(stream_it, sequence_size);
+            stream_it.write_range(std::string_view{" BP;\n"});
+
+            // write sequence in chunks of 60 bp's with a space after 10 bp's
+
+            auto char_sequence = sequence | views::to_char;
+            size_t written_characters{0};
+
+            auto it = stream_it.write_range(char_sequence | views::take(10)); // write first 10 characters
+            stream_it = ' ';
+
+            while (it != char_sequence.end())
```

Comment:
Do not rely on member functions when you are not sure about the type. We make no promises regarding what `views::to_char` returns so it's better to use `std::ranges::begin|end|...`.

Reply:
This affects quite a few places!